utils/obsproc/IcecAmsr2Ioda.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <ctime>
+#include <iostream>
+#include <map>
+#include <netcdf>    // NOLINT (using C API)
+#include <string>
+#include <vector>
+
+#include "eckit/config/LocalConfiguration.h"
+
+#include <Eigen/Dense>    // NOLINT
+
+#include "ioda/Group.h"
+#include "ioda/ObsGroup.h"
+
+#include "NetCDFToIodaConverter.h"
+
+namespace gdasapp {
+
+  class IcecAmsr2Ioda : public NetCDFToIodaConverter {
+   public:
+    explicit IcecAmsr2Ioda(const eckit::Configuration & fullConfig, const eckit::mpi::Comm & comm)
+      : NetCDFToIodaConverter(fullConfig, comm) {
+      variable_ = "seaIceFraction";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by the AMSR2" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get the number of obs in the file
+      int dimxSize = ncFile.getDim("Number_of_X_Dimension").getSize();
+      int dimySize = ncFile.getDim("Number_of_Y_Dimension").getSize();
+      int dimTimeSize = ncFile.getDim("Time_Dimension").getSize();
+      int nobs = dimxSize * dimySize;
+      int ntimes = dimxSize * dimySize * dimTimeSize;
+
+      // Set the int metadata names
+      std::vector<std::string> intMetadataNames = {};
+      // std::vector<std::string> intMetadataNames = {"pass", "cycle", "mission"};
+
+      // Set the float metadata name
+      std::vector<std::string> floatMetadataNames = {};
+      // std::vector<std::string> floatMetadataNames = {"mdt"};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);

```

Comment:
```suggestion
      gdasapp::IodaVars iodaVars(nobs, {}, {});
```
and remove lines 42-48